feat: adds additional commit types

beauraines opened the pull request:
Adds `perf` as a minor release trigger and makes the `cli.js` executable

beauraines-aimconsulting committed:
chore: changes cli executable bit

beauraines-aimconsulting committed:
feat: adds support for Angular commit types

Adds additional commit types from https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type

beauraines commented:
This may need additional linting after #1  is merged.

beauraines commented:
@saintedlama  This may be where you can enable my user to run the workflows without me being a collaborator.

![Screenshot 2023-03-04 at 6 38 25 PM](https://user-images.githubusercontent.com/2186673/222938652-9c84d438-bee7-4f62-8a41-ab597259215d.png)


I really like what you built with this package. Its simple and I can wrap my head around it! I'm pulling it into all of my projects slowly!

beauraines committed:
build: removes test from CI job

There are no unit tests configured

saintedlama commented on cli.js:
```diff
@@ -35,10 +35,10 @@ function whatBump(commits) {
       shouldRelease = true;
       minor++;
-    } else if (commit.type === "fix") {
+    } else if (commit.type === "build" ||  "chore" || "ci" || "docs" || "fix" || "style" || "refactor" || "test") {
```
the idea in this minimal module is to prevent releases for build, chore, ci, docs, style, refactoring, and test changes. Meaning those changes do not require a new version to be published. I think it does not make sense to publish a package for `build`, `chore`, `ci`, `docs`, `style`, `test` or `refactor`, ... changes

beauraines replied:
I've removed these additional type to be inline with the intended design.

saintedlama commented on cli.js:
```diff
@@ -35,10 +35,10 @@ function whatBump(commits) {
       shouldRelease = true;
       major++;
-    } else if (commit.type === "feat") {
+    } else if (commit.type === "feat" || "perf") {
```
👍

beauraines commented:
Got it!  I'll pull that part out. Work has been crazy, so I'll probably do
it this weekend.

--
Beau Raines
***@***.***
425.654.4663 (voice and sms)
beau.raines (skype)
LinkedIn www.linkedin.com/in/beauraines/

On Sun, Mar 12, 2023, 11:02 PM Christoph Walcher ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In cli.js
> <https://github.com/saintedlama/should-release/pull/2#discussion_r1133471596>
> :
>
> >        shouldRelease = true;
>        minor++;
> -    } else if (commit.type === "fix") {
> +    } else if (commit.type === "build" ||  "chore" || "ci" || "docs" || "fix" || "style" || "refactor" || "test") {
>
> the idea in this minimal module is to prevent releases for build, chore,
> ci, docs, style, refactoring, and test changes. Meaning those changes do
> not require a new version to be published. I think it does not make sense
> to publish a package for build, chore, ci, docs, style, test or refactor,
> ... changes
> ------------------------------
>
> In cli.js
> <https://github.com/saintedlama/should-release/pull/2#discussion_r1133471715>
> :
>
> > @@ -35,10 +35,10 @@ function whatBump(commits) {
>      if (commit.notes.length > 0) {
>        shouldRelease = true;
>        major++;
> -    } else if (commit.type === "feat") {
> +    } else if (commit.type === "feat" || "perf") {
>
> 👍
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/saintedlama/should-release/pull/2#pullrequestreview-1336288342>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAQV3MP3KDK2RFWDEWSYSVLW32Z57ANCNFSM6AAAAAAVP4NJZU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

beauraines committed:
Merge branch 'main' into additional-patches

beauraines committed:
chore: linting

beauraines committed:
fix: Cleans up list of patch types

Removes build, chore, ci, docs, style, test or refactor to keep this inline with the projects goals.

beauraines committed:
chore: disables linting for major release logic
